File: 2478-longest-nice-subarray/longest-nice-subarray.py
class Solution:
    def longestNiceSubarray(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        length = 0
        left = 0

        for right, i in enumerate(nums):
            while left < right and (length & i) != 0:
                length ^= nums[left]
                left += 1
            length |= i
            res = max(res, right - left + 1)

        return res

File: 2478-longest-nice-subarray/longest-nice-subarray.java
class Solution {
    public int longestNiceSubarray(int[] nums) {
        int n = nums.length;
        int left = 0;
        int mask = 0;
        int ans = 0;
        
        for (int right = 0; right < n; right++) {
            // While there is a conflict with nums[right], shrink the window.
            while ((mask & nums[right]) != 0) {
                // Remove the leftmost element's bits from the mask.
                mask &= ~nums[left];
                left++;
            }
            // Add the current number to the mask.
            mask |= nums[right];
            // Update the maximum length.
            ans = Math.max(ans, right - left + 1);
        }
        
        return ans;   
    }
}